blobstore/cli/clustermgr/diskmgr.go

```diff
@@ -202,3 +223,125 @@ func openDiskTable(db *normaldb.NormalDB) (*normaldb.DiskTable, error) {
 	}
 	return tbl, nil
 }
+
+func cmdOfflineDisk(c *grumble.Context) error {
+	ctx := common.CmdContext()
+	cmClient := newCMClient(c.Flags)
+	diskid := args.DiskID(c.Args)
+	if diskid <= 0 {
+		return errors.New("invalid common args")
+	}
+
+	if !common.Confirm(fmt.Sprintf("offline disk %d?\n", diskid)) {
+		return nil
+	}
+
+	verbose := config.Verbose() || flags.Verbose(c.Flags)
+	if verbose {
+		fmt.Printf("set disk %d readonly\n", diskid)
+	}
+	if err := readonlyAndDropDisk(ctx, cmClient, diskid); err != nil {
+		return err
+	}
+	fmt.Printf("start to offline disk %d in the background\n", diskid)
+
+	return nil
+}
+
+func cmdOfflineAllDisks(c *grumble.Context) error {
+	ctx := common.CmdContext()
+	cmClient := newCMClient(c.Flags)
+	nodeHost := args.NodeHost(c.Args)
+	if len(nodeHost) <= 0 {
+		return errors.New("invalid args, node host is empty")
+	}
+
+	disks := make([]*blobnode.DiskInfo, 0)
+
+	// list all disks on the node
+	listOptionArgs := &clustermgr.ListOptionArgs{
+		Host:   nodeHost,
+		Status: proto.DiskStatusNormal, // only list normal disks
+	}
+	if listOptionArgs.Marker <= proto.InvalidDiskID {
+		listOptionArgs.Marker = proto.DiskID(1)
+	}
+	for listOptionArgs.Marker > proto.InvalidDiskID {
```

Comment:
just `for` ?  `listOptionArgs.Marker` never be 0.

Reply:
fixed